k3s: fix criteria for removing packages from unstable (#305089)
NixOS/nixpkgs: pkgs/applications/networking/cluster/k3s/README.md

## K3s upkeep for nixpkgs maintainers

> * After every nixos release, K3s maintainers should remove from `nixos-unstable` all K3s versions that exist in `nixos-stable` but the latest version (which is necessary to avoid coupling system upgrade to k3s upgrade).

Reviewer: suggestion(non-blocking): simpler English:

```suggestion
* After every nixos release, K3s maintainers should remove from `nixos-unstable` all K3s versions that exist in `nixos-stable` except the latest version (to allow decoupling system upgrade from k3s upgrade).
```
